### PR TITLE
[Feature][QGIS Server] Add WMS INSPIRE Capabilities

### PR DESCRIPTION
**Pull request by rldhont**
In the project properties the user can:
- activate INSPIRE capabilities
- select language of the service, 24 EU official languages + 5 regionals
- choose the scenario for service metadata and specify them

The WMS 1.3.0 capabilities reflects the INSPIRE configuration.

### CODE REVIEW

File: src/app/qgsprojectproperties.cpp [hunk 1]
```diff
@@ -334,6 +334,77 @@ QgsProjectProperties::QgsProjectProperties( QgsMapCanvas* mapCanvas, QWidget *pa
...
+  bool addWMSInspire = QgsProject::instance()->readBoolEntry( "WMSInspire", "/activated" );
+  if ( addWMSInspire )
+  {
+    mWMSInspire->setChecked( addWMSInspire );
+    QString inspireLanguage = QgsProject::instance()->readEntry( "WMSInspire", "/language", "" );
+    int inspireLanguageIndex = mWMSInspireLanguage->findData( inspireLanguage );
+    mWMSInspireLanguage->setCurrentIndex( inspireLanguageIndex );
+
+    QString inspireMetadataUrl = QgsProject::instance()->readEntry( "WMSInspire", "/metadataUrl", "" );
+    if ( inspireMetadataUrl != "" )
```
Comment (nyalldawson) [inline src/app/qgsprojectproperties.cpp]: FYI - it's much more efficient to test with `!inspireMetadataUrl.isEmpty()`. It avoids unnecessary QString creation and comparison and replaces it with an inexpensive check. There's a few places in this PR which this applies to, eg qgswmsprojectparser.cpp:99

Comment (rldhont) [inline src/app/qgsprojectproperties.cpp]: Thanks, I'll update my code and probably fixed some others.

**Comment by mhugent** [comment]
Thanks for implementing this enhancement, it will be important for many users.

The code looks good to me. In the project properties gui, you could synchronize the checkboxes for scenario 1 and 2 such that selecting one deselects the other.

Somewhen in the future, it will be nice to have a possibility to enter / deliver the metadata fields in several languages.

**Comment by rldhont** [comment]
Hi @mhugent 

Code is updated, I added slots between scenario1 and scenario2.

For the future, I thought about to use multi-projects with different languages but same layer names.

**Pull request closed without merging**